### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by jimt**
I would also hyphenate a lot more words than you have.  Everything from open-source to 8-bit, low-pitched, low-order, fixed-order, post-filtering, time-varying, etc.  I am guessing this is a difference between American and Australian English, so I've tried to resist.

Thank you for taking the time to write this! I may finally have a better grasp of the techniques.

### CODE REVIEW

File: doc/codec2.tex [hunk 1]
```diff
@@ -186,21 +186,21 @@ \subsection{Codec 2 Encoder and Decoder}
...
+Often the errors interact, for example the fine pitch error shown above will mean the amplitude estimates are a little bit off as well. Fortunately, these errors tend to be temporary and are sometimes not even noticeable to the listener - remember this codec is often used for HF/VHF radio where channel noise is part of the normal experience.
  
 \begin{figure}[h]
 \caption{Codec 2 Decoder}
 \label{fig:codec2_decoder}
 \begin{center}
 \begin{tikzpicture}[auto, node distance=2cm,>=triangle 45,x=1.0cm,y=1.0cm,align=center,text width=2cm]
 
-\node [input] (rinput) {};
+\node [input] (input) {};
```
Comment (drowe67) [inline doc/codec2.tex:203]: @jimt - this is a logical internal name to ID a part of the figure, ie `rinput` doesn't appear anywhere in the text. Changing the name without changing the name of Tikz lines using it below will break the doc build.  Suggest you revert these sorts of changes.

I've enabled the automated tests so you can tell if anything you change breaks the doc build.  or you can run the doc build yourself if you have an Ubuntu machine  (instructions in `doc/Makefile`).

**Comment by drowe67** [comment]
Hi @jimt - thanks for your fine proof reading (yet again) :slightly_smiling_face: 

Can you pls merge your PR with the latest main branch?  @timw has just fixed the automated build test for the doc, so when you push, it will run a test to build the doc.

Re hyphenation - I'm actually not sure what to do, I'm just guessing.  Is there an external reference or style guide we can refer too?  I've also adopted .au spelling (e.g. quantisation) for better or for worse.  That was just a random choice too.

File: doc/codec2.tex [hunk 2]
```diff
@@ -218,19 +218,19 @@ \subsection{Codec 2 Encoder and Decoder}
...
 
-Vector Quantisation can only represent what is present in the tables, so if it sees anything unusual (for example a different microphone frequency response or background noise), the quantisation can become very rough and speech quality poor.  We train the tables at design time using a database of speech samples and a training algorithm - an early form of machine learning.
+Vector Quantisation can only represent what is present in the tables, so if it sees anything unusual (for example, a different microphone frequency response or background noise), the quantization can become very rough and speech quality poor.  We train the tables at design time using a database of speech samples and a training algorithm - an early form of machine learning.
```
Comment (drowe67) [inline doc/codec2.tex:234]: :thinking: - so this is where I guess we need to make a call on spelling conventions, e.g. The ~Queens~ Kings English, or some other locale.  Actually I haven't checked quantisation is in an Australian dictionary .... just assuming the s/s/z thing.

File: doc/codec2.tex [hunk 3]
```diff
@@ -345,7 +345,7 @@ \subsection{Sinusoidal Analysis}
 \end{equation}
 The DFT indexes $a_m, b_m$ select the band of $S_w(k)$ containing the $m$-th harmonic; $r$ maps the harmonic number $m$ to the nearest DFT index, and $\lfloor x \rceil$ is the rounding operator. This method of estimating $A_m$ is relatively insensitive to small errors in $F0$ estimation and works equally well for voiced and unvoiced speech.  Figure $\ref{fig:hts2a_time}$ plots $S_w$ (blue) and $\{A_m\}$ (red) for a sample frame of female speech.
 
-The phase is sampled at the centre of the band.  For all practical Codec 2 modes the phase is not transmitted to the decoder so does not need to be computed.  However speech synthesised using the phase is useful as a control during development, and is available using the \emph{c2sim} utility.
+The phase is sampled at the centre of the band.  For all practical Codec 2 modes, the phase is not transmitted to the decoder, so it does not need to be computed.  However, speech synthesized using the phase is useful as a control during development and is available using the \emph{c2sim} utility.
 
 \subsection{Sinusoidal Synthesis}
```
Comment (drowe67) [inline doc/codec2.tex:350]: Another s/z question.

File: doc/codec2.tex (hunk 1 above)
Comment (jimt) [reply doc/codec2.tex:203]: Oops.  I didn't intend to change that.

**Comment by jimt** [comment]
> Re hyphenation - I'm actually not sure what to do, I'm just guessing. Is there an external reference or style guide we can refer too?

I go by my poorly remembered Chicago Manual of Style rules about compound modifiers, but I see [the guidance has changed a bit](https://business.columbia.edu/identity/editorial-style/dashes-and-hyphens) over the years. I suppose it is best to wait for a real editor to review your document.

**Comment by drowe67** [comment]
Thanks @jimt - the `test_codec2_doc` test ran OK.  Are you able to build the pdf and commit it?  Instructions are in `doc/Makefile`. Or let me know if you would like me to.

**Comment by jimt** [comment]
Please do it for me.

When I tried there were warnings about duplicate table and figure names (in addition to more typical complaints about overfull and underfill boxes). And one `no ; in font nullfont` so I don't have confidence in my output.

**Comment by drowe67** [comment]
Yes it's very verbose. Generally if the PDF gets created all is well.  I'll push the latest and greatest in a  few days, I'm off on a  short vacation :boat:

**Comment by drowe67** [comment]
Hi @jimt,  so I pulled your branch and rebuilt the PDF (it looks fine) but I can't push to your repository (it keeps requesting my user name).  I think you may need to [allow changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).  This sort of thing is at the edge of my git skills, so failing that we can just merge your changes and I will push the new PDF to `main`.

**Comment by tmiw** [comment]
> Hi @jimt, so I pulled your branch and rebuilt the PDF (it looks fine) but I can't push to your repository (it keeps requesting my user name). I think you may need to [allow changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork). This sort of thing is at the edge of my git skills, so failing that we can just merge your changes and I will push the new PDF to `main`.

I was able to do it via GitHub Desktop. Presumably there's a way to do it via the command prompt as well but /shrug.

**Comment by drowe67** [comment]
> > Hi @jimt, so I pulled your branch and rebuilt the PDF (it looks fine) but I can't push to your repository (it keeps requesting my user name). I think you may need to [allow changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork). This sort of thing is at the edge of my git skills, so failing that we can just merge your changes and I will push the new PDF to `main`.
> 
> I was able to do it via GitHub Desktop. Presumably there's a way to do it via the command prompt as well but /shrug.

Thanks !

> BTW I just realized that the current Makefile doesn't work on macOS, but that's something that can be fixed outside of this PR.

Cool.  Or it could perhaps be camke

**Comment by drowe67** [comment]
Oops fumbled the Git buttons! and accidentally closed :slightly_smiling_face:

**Pull request merged**